ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +268,155 @@ void PowerSystemNode::injectFault (const string& fault_name,
     {
       ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
       wattage += hpd_wattage;
-    } else
+    }
+  }
+}
+
+void PowerSystemNode::injectCustomFault(bool& fault_activated,
+                                        const PrognoserVector& sequence,
+                                        size_t& index,
+                                        double& wattage,
+                                        double& voltage,
+                                        double& temperature)
+{
+  static string saved_fault_directory = "N/A";
+  string current_fault_directory;
+  string designated_file;
+  static string saved_file;
+  static bool custom_fault_ready = false;
+  static bool custom_warning_displayed = false;
+  static bool end_fault_warning_displayed = false;
+  bool fault_enabled = false;
+
+  // Do nothing unless the specified fault has been injected.
+  if (! ros::param::getCached("/faults/activate_custom_fault", fault_enabled)) {
+    return;
+  }
+
+  if (!fault_activated && fault_enabled)
+  {
+    // Only continue if no other faults have been injected.
+    ROS_INFO_STREAM("Attempting custom fault activation...");
+    if (HPD_FAULT_ACTIVATED)
+    {
+      if (!custom_warning_displayed)
+      {
+        ROS_WARN_STREAM("Cannot activate custom fault: high power draw already used this simulation.");
+        custom_warning_displayed = true;
+      }
+    }
+    else
+    {
+      // Get user-entered file directory
+      if (! ros::param::getCached("/faults/custom_fault_profile", designated_file))
+      {
+        ROS_ERROR("Failed to fetch custom fault name.");
```

Comment:
I presume this should never happen.  In any case, the check is fine but maybe call it "custom fault profile name".

Reply:
I don't think it should happen either; this check was simply part of the old code we're repurposing. I can remove it if you'd like, otherwise I'll modify the error message.

Reply:
Leave the check in (adds robustness), just modify the message, thanks!